src/models/campaign.ts

```diff
@@ -176,6 +177,15 @@ export default class Campaign {
             metadataConfig.dataElementGroupCodeForAntigens,
         ]);
 
+        const dashboardId = await this.db.createDashboard(
```

Comment:
I realize this is something we have not discussed yet: whenever possible, we should create all objects (dataset, analytic objects, dashboard, ...) in a single POST /metadata request (`this.db.postMetadata`). That's not only because it's faster, but also because we are wrapping everything in a transaction, all or nothing is created. Otherwise, we can create a dashboard with charts, then the dataset save may fail, and we end up with dangling objects. 

The trick for this is to set the UID for new objects in our JS client code (generateUid). You can see that's what we are doing for dataset. I am not sure if this works in your case, I hope it does, it makes the whole process more robust.

Reply:
I'll try this

Reply:
works!